compiler/ras/DebugCounter.hpp

```diff
@@ -229,7 +230,20 @@ class DebugCounterAggregation : public DebugCounterBase
 
 class DebugCounterGroup
    {
-   CS2::HashTable<const char*, DebugCounter*, TRPersistentMemoryAllocator> _countersHashTable;
+   struct debugCounterNameComp
+      {
+      bool operator()(const char* lhs, const char* rhs) const
+         {
+         while (*lhs == *rhs)
+            {
+            if (*lhs==0) return false;
```

Comment:
I think *rhs==0 is implied by the while condition, so in this case the two char arrays are the same. We want to return true only if the lhs char array is greater than the rhs char array.